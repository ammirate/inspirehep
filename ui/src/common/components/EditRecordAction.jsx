import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ListItemAction from './ListItemAction';
import IconText from './IconText';
import EventTracker from './EventTracker';
import ExternalLink from './ExternalLink';

import { EDIT_LITERATURE, EDIT_JOB } from '../routes';


const pidTypeToEditRoutePrefix = {
  literature: EDIT_LITERATURE,
  jobs: EDIT_JOB,
}

class EditRecordAction extends Component {

  render() {
    const { pidType, pidValue } = this.props;

    return (
      <ListItemAction>
        <EventTracker eventId="Edit">
          <ExternalLink href={`${pidTypeToEditRoutePrefix[pidType]}/${pidValue}`}>
            <IconText text="edit" type="edit" />
          </ExternalLink>
        </EventTracker>
      </ListItemAction>
    );
  }
}

EditRecordAction.propTypes = {
    pidType: PropTypes.oneOf(['literature', 'jobs']).isRequired,
    pidValue: PropTypes.number.isRequired,
};

export default EditRecordAction;
