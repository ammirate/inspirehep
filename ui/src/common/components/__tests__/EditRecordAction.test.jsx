import React from 'react';
import { shallow } from 'enzyme';
import EditRecordAction from '../EditRecordAction';

describe('EditRecordAction', () => {
  it('renders edit button with pidType literature and pidValue', () => {
    const wrapper = shallow(
      <EditRecordAction pidType='literature' pidValue={1}/>
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders edit button with pidType jobs and pidValue', () => {
    const wrapper = shallow(
      <EditRecordAction pidType='jobs' pidValue={1}/>
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
