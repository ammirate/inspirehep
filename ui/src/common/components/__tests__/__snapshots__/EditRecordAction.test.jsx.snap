// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecordAction renders edit button with pidType jobs and pidValue 1`] = `
<span
  className="__ListItemAction__"
>
  <EventTracker
    eventId="Edit"
    eventPropName="onClick"
    extractEventArgsToForward={null}
  >
    <ExternalLink
      href="/submissions/jobs/1"
    >
      <IconText
        text="edit"
        type="edit"
      />
    </ExternalLink>
  </EventTracker>
</span>
`;

exports[`EditRecordAction renders edit button with pidType literature and pidValue 1`] = `
<span
  className="__ListItemAction__"
>
  <EventTracker
    eventId="Edit"
    eventPropName="onClick"
    extractEventArgsToForward={null}
  >
    <ExternalLink
      href="/workflows/edit_article/1"
    >
      <IconText
        text="edit"
        type="edit"
      />
    </ExternalLink>
  </EventTracker>
</span>
`;
